python-client/tests/test_push.py

```diff
@@ -126,18 +128,35 @@ def test_slicing_function_multi(enron_model, enron_data):
             assert isinstance(push.slicing_function, QueryBasedSliceFunction)
 
 
-def test_test_function_multi(enron_model, enron_data):
+def test_test_function_and_output(enron_model, enron_data):
+
+    # Define a list of push types and corresponding functions
+    push_types = [
+        (giskard.push.ContributionPush, create_contribution_push),
+        (giskard.push.PerturbationPush, create_perturbation_push),
+        (giskard.push.OverconfidencePush, create_overconfidence_push),
+        (giskard.push.BorderlinePush, create_borderline_push),
+    ]
+
+    # Initialize dictionaries to store outputs
+    outputs = {push_type: [] for push_type, _ in push_types}
+
+    # Loop through the range and create push outputs
     for i in range(50):
-        push_list = [
-            create_contribution_push(enron_model, enron_data, enron_data.df.iloc[[i]]),
-            create_perturbation_push(enron_model, enron_data, enron_data.df.iloc[[i]]),
-            create_overconfidence_push(enron_model, enron_data, enron_data.df.iloc[[i]]),
-            create_borderline_push(enron_model, enron_data, enron_data.df.iloc[[i]]),
-        ]
-        for push in push_list:
-            if push is not None:
-                for test in push.tests:
-                    assert isinstance(test(), GiskardTest)
+        row = enron_data.df.iloc[[i]]
+        for push_type, create_push_function in push_types:
+            push_output = create_push_function(enron_model, enron_data, row)
+            if push_output is not None:
+                outputs[push_type].append(push_output)
+
+    # Check if push output has the correct type
+    # Check if the tests created by push are instance of GiskardTest
+    for push_type, push_outputs in outputs.items():
+        if push_outputs:
```

Comment:
related to the comment above, if we know that a specific row should result in a specific set of pushes we shouldn't use `if push_outputs`, but rather `assert len(push_outputs)>0`